### PR TITLE
Develop lvuidf 27 implement admin part app logic

### PR DESCRIPTION
**Pull request by khomyakov**
https://ssu-jira.softserveinc.com/browse/LVUIDF-26
https://ssu-jira.softserveinc.com/browse/LVUIDF-27

Implement design of admin part of application
Agree one design. Should be simple and convenients
Create separated namespace for 'admin' part of application
Almost each model will have 2 controllers. One - for Admin part of application and another one for users.

To review:
- [x] @rtriska
- [ ] @vyshnevska
- [ ] @forecome 

To merge:
- [ ] @izagor
- [x] @forecome

### CODE REVIEW

File: app/assets/javascripts/app.js.jsx [hunk 1]
```diff
@@ -5,11 +5,14 @@ var Bb = {
...
 
+var RouteHandler = ReactRouter.RouteHandler,
+    Link = ReactRouter.Link;
+
 // main React component
 var App = React.createClass({
   render: function () {
     return (
-      <ReactRouter.RouteHandler />
+	        <RouteHandler />
```
Comment (forecome) [inline app/assets/javascripts/app.js.jsx]: What is purpose of these changes?

File: app/assets/javascripts/components/manager/breadcrumbs.js.jsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+var Breadcrumbs = React.createClass({
+  render: function() {
+    var PageName = "Page";
+    return (
+      <div className="breadcrumb">
+        <li>
+          <a href="#">Home</a>
+        </li>
+        <li className="active">{PageName}</li>
```
Comment (forecome) [inline app/assets/javascripts/components/manager/breadcrumbs.js.jsx]: Why did you not write just `Page`? For what you need this variable?

File: app/assets/javascripts/components/manager/notifications.js.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      <div className="panel panel-default">
+        <div className="panel-heading">
+          <i className="glyphicon glyphicon-tasks"></i>
+          <span> Notifications</span>
+        </div>
+        <div className="panel-body">
+          <div className="list-group">
+            <a className="list-group-item" href="#">
+              New Comments
+              <span className="badge">14</span>
```
Comment (forecome) [inline app/assets/javascripts/components/manager/notifications.js.jsx]: Counters are hardcoded. Please, render proper number here and below

File: app/assets/javascripts/components/manager/showcomponent.js.jsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+var ShowComponent = React.createClass({
+  render: function() {
+    return (
+      <div></div>
```
Comment (forecome) [inline app/assets/javascripts/components/manager/showcomponent.js.jsx]: Do you need this empty component?

File: app/assets/javascripts/components/manager/users/articlesviews.js.jsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            <p className="list-group-item-text">Donec id elit non mi porta gravida at eget metus. Maecenas sed diam eget risus varius blandit.</p>
+          </a>
+        </div>
+        </div>
+      </div>
+    );
+  }
+});
+
+var ArticlesAdd = React.createClass({
```
Comment (forecome) [inline app/assets/javascripts/components/manager/users/articlesviews.js.jsx]: Maybe will be better to write two files instead of describing all classes in one file?

File: app/assets/javascripts/components/manager/users/articlesviews.js.jsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    return (
+      <div className="row">
+        <div className="col-md-12">
+        <div class="list-group">
+          <a href="#" className="list-group-item">
+            <div className='btn-group pull-right'>
+                <button className="btn btn-default btn-sm">Edit</button>
+                <button className="btn btn-sm btn-danger">Delete</button>
+            </div>
+            <h4 className="list-group-item-heading">Article 1</h4>
```
Comment (forecome) [inline app/assets/javascripts/components/manager/users/articlesviews.js.jsx]: Hardcoded dummy data. Is it ok?

File: app/assets/javascripts/components/manager/users/usersviews.js.jsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+              <th>Email</th>
+              <th>Name</th>
+              <th>Role</th>
+              <th>Options</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td>1</td>
+              <td>lolpop32@giraffe.com</td>
```
Comment (forecome) [inline app/assets/javascripts/components/manager/users/usersviews.js.jsx]: Another dummy data

File: app/assets/javascripts/components/manager/users/usersviews.js.jsx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+              </td>
+            </tr>
+          </tbody>
+        </table>
+      </div>
+    );
+  }
+});
+
+var UserAdd = React.createClass({
```
Comment (forecome) [inline app/assets/javascripts/components/manager/users/usersviews.js.jsx]: Please, separate classes in different files

File: app/controllers/manager_controller.rb [hunk 9]
```diff
@@ -0,0 +1,3 @@
+class ManagerController < ApplicationController
+  layout 'manager/manager'
```
Comment (forecome) [inline app/controllers/manager_controller.rb]: Please, store layouts in `views/layouts` directory

File: app/helpers/manager_helper.rb [hunk 10]
```diff
@@ -0,0 +1,2 @@
+module ManagerHelper
```
Comment (forecome) [inline app/helpers/manager_helper.rb]: Please, delete this empty helper

File: spec/controllers/manager_controller_spec.rb [hunk 11]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe ManagerController, type: :controller do
```
Comment (forecome) [inline spec/controllers/manager_controller_spec.rb]: Please, remove empty spec file

File: spec/helpers/manager_helper_spec.rb [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+# the ManagerHelper. For example:
+#
+# describe ManagerHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe ManagerHelper, type: :helper do
```
Comment (forecome) [inline spec/helpers/manager_helper_spec.rb]: Please, remove empty helper spec

File: app/assets/javascripts/app.js.jsx (hunk 1 above)
Comment (khomyakov) [inline app/assets/javascripts/app.js.jsx]: compatibility issues after rebase

File: app/assets/javascripts/components/manager/breadcrumbs.js.jsx (hunk 2 above)
Comment (khomyakov) [inline app/assets/javascripts/components/manager/breadcrumbs.js.jsx]: deprecated view, i forgot to remove

File: spec/helpers/manager_helper_spec.rb (hunk 12 above)
Comment (khomyakov) [inline spec/helpers/manager_helper_spec.rb]: ok

File: spec/controllers/manager_controller_spec.rb (hunk 11 above)
Comment (khomyakov) [inline spec/controllers/manager_controller_spec.rb]: ok

File: app/assets/javascripts/components/manager/showcomponent.js.jsx (hunk 4 above)
Comment (khomyakov) [inline app/assets/javascripts/components/manager/showcomponent.js.jsx]: removed

File: app/assets/javascripts/components/manager/users/articlesviews.js.jsx (hunk 5 above)
Comment (khomyakov) [inline app/assets/javascripts/components/manager/users/articlesviews.js.jsx]: it's placeholder for demo, should be replaced by proper crud views (not my task)

File: app/assets/javascripts/components/manager/users/articlesviews.js.jsx (hunk 6 above)
Comment (khomyakov) [inline app/assets/javascripts/components/manager/users/articlesviews.js.jsx]: it's placeholder for demo, should be replaced by proper crud views (not my task)

File: app/assets/javascripts/components/manager/users/usersviews.js.jsx (hunk 7 above)
Comment (khomyakov) [inline app/assets/javascripts/components/manager/users/usersviews.js.jsx]: it's placeholder for demo, should be replaced by proper crud views (not my task)

File: app/assets/javascripts/components/manager/users/usersviews.js.jsx (hunk 8 above)
Comment (khomyakov) [inline app/assets/javascripts/components/manager/users/usersviews.js.jsx]: it's placeholder for demo, should be replaced by proper crud views (not my task)

Comment (khomyakov) [inline app/assets/javascripts/components/manager/users/usersviews.js.jsx]: fixed

File: app/controllers/manager_controller.rb (hunk 9 above)
Comment (khomyakov) [inline app/controllers/manager_controller.rb]: fixed

File: app/helpers/manager_helper.rb (hunk 10 above)
Comment (khomyakov) [inline app/helpers/manager_helper.rb]: done

File: app/assets/javascripts/app.js.jsx (hunk 1 above)
Comment (forecome) [inline app/assets/javascripts/app.js.jsx]: Please, revert these changes

File: app/assets/javascripts/components/manager/users/usersviews.js.jsx (hunk 8 above)
Comment (forecome) [inline app/assets/javascripts/components/manager/users/usersviews.js.jsx]: Ok, but any way, write two different view in two different files

File: app/assets/javascripts/components/manager/users/articlesviews.js.jsx (hunk 5 above)
Comment (forecome) [inline app/assets/javascripts/components/manager/users/articlesviews.js.jsx]: Please, write one view per one file

File: app/controllers/manager_controller.rb [hunk 13]
```diff
@@ -0,0 +1,6 @@
+class ManagerController < ApplicationController
+  layout 'manager'
+
+  before_action :authenticate_user!
+  before_filter :require_admin
```
Comment (forecome) [inline app/controllers/manager_controller.rb]: `before_filter` is deprecated

**Comment by yurko** [comment]
just a remark: you should not add ```ready```, it's the right of person who merges

File: app/assets/javascripts/app.js.jsx [hunk 14]
```diff
@@ -5,11 +5,14 @@ var Bb = {
   Helpers: {}
 };
 
+var RouteHandler = ReactRouter.RouteHandler,
+    Link = ReactRouter.Link;
```
Comment (forecome) [inline app/assets/javascripts/app.js.jsx]: You reverted changes below, but what about this vars? They are not used anywhere.

File: app/controllers/application_controller.rb [hunk 15]
```diff
@@ -10,4 +10,11 @@ def index
     render text: '', layout: true
   end
 
+  def require_admin
+    unless current_user && current_user.role == 'admin'
```
Comment (forecome) [inline app/controllers/application_controller.rb]: Please, use guard clause (https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals)

File: app/assets/javascripts/components/manager/users/usersdisplay.js.jsx [hunk 16]
```diff
@@ -0,0 +1,51 @@
+var UsersDisplay = React.createClass({
```
Comment (rtriska) [inline app/assets/javascripts/components/manager/users/usersdisplay.js.jsx]: Hardcoded data in component? Why do we need this?

File: app/assets/javascripts/components/manager/users/articlesdisplay.js.jsx [hunk 17]
```diff
@@ -0,0 +1,38 @@
+var ArticlesDisplay = React.createClass({
```
Comment (rtriska) [inline app/assets/javascripts/components/manager/users/articlesdisplay.js.jsx]: Why do we need hardcoded data?

File: app/assets/javascripts/components/manager/users.js.jsx [hunk 18]
```diff
@@ -0,0 +1,24 @@
+var Users = React.createClass({
+  getInitialState: function(){
+    return {checked: 'display', pagetitle: 'Users'};
+  },
+  render: function() {
+    var swicher;
+      if (this.state.checked == 'display') {
```
Comment (rtriska) [inline app/assets/javascripts/components/manager/users.js.jsx]: check if next works
```javascript
var switcher = this.state.checked == 'display' ? <UsersDisplay /> : <UserAdd />
```
In case yes, please refactor

File: app/assets/javascripts/components/manager/title.js.jsx [hunk 19]
```diff
@@ -0,0 +1,20 @@
...
+            <button id='add' onClick={this.handleClick} className='btn btn-default btn-md'>Add</button>
+          </div>
+        </div>
+      </div>
+    );
+	},
+  handleClick: function(e) {
+    this.props.onTitleLinkClick(e.target.id);
+  },
+});
```
Comment (rtriska) [inline app/assets/javascripts/components/manager/title.js.jsx]: add new line

File: app/assets/javascripts/manager.js [hunk 20]
```diff
@@ -0,0 +1,4 @@
+// Place all the behaviors and hooks related to the matching controller here.
+// All this logic will automatically be available in application.js.
+//= require jquery
+//= require bootstrap-sprockets
```
Comment (rtriska) [inline app/assets/javascripts/manager.js]: add new line

File: app/assets/javascripts/components/manager/articles.js.jsx [hunk 21]
```diff
@@ -0,0 +1,24 @@
+var Articles = React.createClass({
+  getInitialState: function(){
+    return {checked: 'display', pagetitle: 'Articles'};
+  },
+  render: function() {
+    var swicher;
+      if (this.state.checked == 'display') {
```
Comment (rtriska) [inline app/assets/javascripts/components/manager/articles.js.jsx]: See example with 'switcher' below... In 'app/assets/javascripts/components/manager/users.js.jsx'

**Comment by rtriska** [comment]
Looks good :+1:

**Comment by yurko** [comment]
@forecome, please revert this merge. @khomyakov added component ```Articles``` named the same as already existing one, and it affected article's views.
